src/Symfony/Component/VarDumper/Command/ServerDumpCommand.php

```diff
@@ -0,0 +1,301 @@
+
+    private $cliDumper;
+    private $htmlDumper;
+    private $fileLinkFormatter;
+    private $projectDir;
+    private $logger;
+    /** @var Filesystem */
+    private $filesystem;
+    private $lastIdentifier;
+
+    public function __construct(CliDumper $cliDumper = null, HtmlDumper $htmlDumper = null, FileLinkFormatter $fileLinkFormatter, $projectDir = null, LoggerInterface $logger = null)
+    {
+        $this->cliDumper = $cliDumper;
+        $this->htmlDumper = $htmlDumper;
+        $this->fileLinkFormatter = $fileLinkFormatter;
+        $this->projectDir = $projectDir;
+        $this->logger = $logger;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('server:dump')
+            ->addOption('output', null, InputOption::VALUE_REQUIRED, 'An file path to save dumped data in html format, or null to display on CLI output.')
+            ->setDescription('Starts a dump server that collects and displays dumps in a single place')
+            ->setHelp(<<<'EOF'
+<info>%command.name%</info> starts a dump server that collects and displays 
+dumps in a single place for debugging you application:
+
+  <info>php %command.full_name%</info>
+
+You can consult dumped data in html format by providing an output file path and opening it in your browser:
+
+  <info>php %command.full_name% --output="dump.html"</info>
+
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $io->title('Symfony Var Dumper Server');
+
+        $server = new DumpServer(null, $this->logger);
+
+        $server->start();
+
+        $io->success(sprintf('Server listening on %s', $server->getHost()));
+
+        if (null !== $outputFile = $input->getOption('output')) {
+            $this->filesystem = new Filesystem();
+            $this->htmlDumper = $this->htmlDumper ?: new HtmlDumper();
+
+            if (!$this->filesystem->exists($outputFile)) {
+                $this->filesystem->touch($outputFile);
+                $this->filesystem->appendToFile($outputFile, sprintf('<style>%s</style><script>%s</script>', self::$style, self::$javascript));
+            }
+
+            $io->comment(sprintf('Consult collected dumps in html format by opening "%s" in your browser.', realpath($outputFile)));
+        } else {
+            $this->cliDumper = $this->cliDumper ?: new CliDumper();
+        }
+
+        $io->comment('Quit the server with CONTROL-C.');
+
+        $server->listen(function (Data $data, array $context, $clientId) use ($io, $outputFile) {
+            if ($outputFile) {
+                $this->displayDumpHtml($data, $context, $outputFile);
+            } else {
+                $this->displayDumpCli($io, $data, $context, $clientId);
+            }
+        });
+    }
+
+    private function displayDumpCli(SymfonyStyle $io, Data $data, array $context, $clientId)
```

Comment:
If I'm creating my own server (like explained in the PR description), I would probably like to be able to reuse the logic in those 2 private methods, right?